flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+/**
+ * A collection of all configuration options that relate to checkpoints
+ * and savepoints.
+ */
+public class CheckpointingOptions {
+
+	// ------------------------------------------------------------------------
+	//  general checkpoint and state backend options
+	// ------------------------------------------------------------------------
+
+	public static final ConfigOption<String> STATE_BACKEND = ConfigOptions
+			.key("state.backend")
+			.noDefaultValue();
+
+	/** The maximum number of completed checkpoint instances to retain.*/
+	public static final ConfigOption<Integer> MAX_RETAINED_CHECKPOINTS = ConfigOptions
+			.key("state.checkpoints.num-retained")
+			.defaultValue(1);
+
+	// ------------------------------------------------------------------------
+	//  Options specific to the file-system-based state backends
+	// ------------------------------------------------------------------------
+
+	/** The default directory for savepoints. Used by the state backends that write
+	 * savepoints to file systems (MemoryStateBackend, FsStateBackend, RocksDBStateBackend). */
+	public static final ConfigOption<String> SAVEPOINT_DIRECTORY = ConfigOptions
+			.key("state.savepoints.dir")
+			.noDefaultValue()
+			.withDeprecatedKeys("savepoints.state.backend.fs.dir");
+
+	/** The default directory used for checkpoints. Used by the state backends that write
+	 * checkpoints to file systems (MemoryStateBackend, FsStateBackend, RocksDBStateBackend). */
+	public static final ConfigOption<String> CHECKPOINTS_DIRECTORY = ConfigOptions
+			.key("state.checkpoints.dir")
+			.noDefaultValue();
+
+	/** Option whether the heap-based key/value data structures should use an asynchronous
+	 * snapshot method. Used by MemoryStateBackend and FsStateBackend. */
+	public static final ConfigOption<Boolean> HEAP_KV_ASYNC_SNAPSHOTS = ConfigOptions
+			.key("state.backend.heap.async")
```

Comment:
Off-topic: The fact that we have "FsStateBackend" and "MemoryStateBackend" is confusing every user. We should only have "HeapStateBackend" and "RocksDBStateBackend", which both checkpoint to a DFS. (And the current "MemoryStateBackend" behaviour could be a switch on "HeapStateBackend"). But I'm afraid it's too late for that. 😩

Reply:
Mentioning it because we have no "heap" backend, technically.

Reply:
+1 to fix the names, separate effort though

Reply:
Yes, I know. I would like to do that as a follow-up once you're done, time permitting.